### PR TITLE
[NOD-1497] Add additional methods to consensus' external API

### PR DESCRIPTION
**Pull request by stasatdaglabs**

### CODE REVIEW

File: domain/consensus/consensus.go [hunk 1]
```diff
@@ -70,42 +79,69 @@ func validateTransactionInContextAndPopulateMassAndFeeVirtualBlockHash() *extern
...
-	panic("implement me")
+	blockInfo := &externalapi.BlockInfo{}
+
+	exists, err := s.blockStatusStore.Exists(s.databaseContext, blockHash)
+	if err != nil {
+		return nil, err
+	}
+	blockInfo.Exists = exists
+	if !exists {
+		return blockInfo, nil
```
Comment (svarogg) [inline domain/consensus/consensus.go:99]: We should probably return something that is not an empty BlockInfo to signal that block doesn't exist (I'd say return nil, but you'll shoot me for such insolent behavior)

File: domain/consensus/consensus.go [hunk 2]
```diff
@@ -71,41 +80,68 @@ func validateTransactionInContextAndPopulateMassAndFeeVirtualBlockHash() *extern
...
 func (s *consensus) GetVirtualSelectedParent() (*externalapi.DomainBlock, error) {
-	panic("implement me")
+	virtualGHOSTDAGData, err := s.ghostdagDataStore.Get(s.databaseContext, model.VirtualBlockHash)
+	if err != nil {
+		return nil, err
+	}
+	return s.GetBlock(virtualGHOSTDAGData.SelectedParent)
 }
 
 func (s *consensus) CreateBlockLocator(lowHigh, highHash *externalapi.DomainHash) (*externalapi.BlockLocator, error) {
```
Comment (svarogg) [inline domain/consensus/consensus.go:141]: typo: lowHigh

File: domain/consensus/processes/syncmanager/syncmanager.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+func (s syncManager) GetHashesBetween(lowHigh, highHash *externalapi.DomainHash) ([]*externalapi.DomainHash, error) {
+	panic("implement me")
+}
+
+func (s syncManager) GetMissingBlockBodyHashes(highHash *externalapi.DomainHash) ([]*externalapi.DomainHash, error) {
+	panic("implement me")
+}
+
+func (s syncManager) CreateBlockLocator(lowHigh, highHash *externalapi.DomainHash) (*externalapi.BlockLocator, error) {
```
Comment (svarogg) [inline domain/consensus/processes/syncmanager/syncmanager.go:27]: lowHigh here too

**Review by svarogg — CHANGES_REQUESTED** [review]

File: domain/consensus/consensus.go (hunk 1 above)
Comment (stasatdaglabs) [reply domain/consensus/consensus.go:99]: We do return blockInfo.Exists = false.

**Review by svarogg — APPROVED** [review]

**Pull request merged**